module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -0,0 +1,99 @@
+## Intoduction to database schemas
+
+A database schema is a blueprint or architecture of how our data will look. It doesn’t hold data itself, but instead describes the shape of the data and how it might relate to other tables or models.
+
+For example, this is a `customer` schema in MongoDB using mongoose.
+
+```js
+const mongoose = require("mongoose");
+
+const customer = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  zipcode: {
+    type: Number,
+  },
+});
+
+module.exports = mongoose.model("Customer", customer);
+```
+
+The same `customer` schema in a MySql database.
+
+```sql
+CREATE TABLE customer (
+ id INT AUTO_INCREMENT PRIMARY KEY,
+ name VARCHAR(50) NOT NULL,
+ zipcode INT(15) default NULL,
+)
+
+```
+
+We can see that the schema definition for the `customer` has a `name` which is a type `string` and a `zipcode` that is of type `number`.
+
+## Introduction to CRUD operations
+
+When we are building APIs, we want our models to provide four basic types of functionality. The model must be able to Create, Read, Update, and Delete resources. Computer scientists often refer to these functions by the acronym `CRUD`. A model should have the ability to perform at most these four functions in order to be complete.
+
+### CRUD and REST
+
+### What is REST?
+
+REST is a set of architectural constraints. API developers can implement REST in a variety of ways. When a client request is made via a RESTful API, it transfers a representation of the state of the resource to the requester or endpoint. This information, or representation, is delivered in one of several formats, JSON is the most generally popular file format to use.
```

Comment:
don't quite like this phrase "transfers a representation of the state of the resource", not sure i even understand it :) i mean i understand what it means but i think it will be unnecessarily complex for students, perhaps find a simpler way to explain